hddm/models/hddm_nn_regression.py

```diff
@@ -0,0 +1,330 @@
+from collections import OrderedDict
+from copy import deepcopy
+import math
+import numpy as np
+import pymc as pm
+import pandas as pd
+from patsy import dmatrix
+#import pickle
+
+import hddm
+from hddm.models import HDDM
+from hddm.keras_models import load_mlp
+from hddm.cnn.wrapper import load_cnn
+import kabuki
+from kabuki import Knode
+from kabuki.utils import stochastic_from_dist
+import kabuki.step_methods as steps
+from functools import partial
+import wfpt
+
+class KnodeRegress(kabuki.hierarchical.Knode):
+    def __init__(self, *args, **kwargs):
+        
+        # Whether or not to keep the regressor trace
+        self.keep_regressor_trace = kwargs.pop('keep_regressor_trace', False)
+        
+        # Initialize kabuke.hierarchical.Knode
+        super(KnodeRegress, self).__init__(*args, **kwargs)
+
+    def create_node(self, name, kwargs, data):
+        reg = kwargs['regressor']
+        
+        # order parents according to user-supplied args
+        # QAF: what's the point of that?
+
+        args = []
+        for arg in reg['params']:
+            for parent_name, parent in kwargs['parents'].items():
+                if parent_name == arg:
+                    args.append(parent)
+
+        parents = {'args': args}
+
+        # Make sure design matrix is kosher
+        dm = dmatrix(reg['model'], data = data)
+        if math.isnan(dm.sum()):
+            raise NotImplementedError('DesignMatrix contains NaNs.')
+
+        def func(args, 
+                 design_matrix = dmatrix(reg['model'], data = data), 
+                 link_func = reg['link_func']):
+            
+            # convert parents to matrix
+            params = np.matrix(args)
+            
+            # Apply design matrix to input data
+            if design_matrix.shape[1] != params.shape[1]:
+                raise NotImplementedError('Missing columns in design matrix. You need data for all conditions for all subjects.')
+            
+            # Get 'predictors' ( wfpt model parameters according to current regression parameterization)
+            predictor = link_func(pd.DataFrame((design_matrix * params).sum(axis = 1), # compute regression matrix algebra
+                                                index = data.index)) # QAF why data.index, is this necessary ?
+
+            return pd.DataFrame(predictor, index = data.index)
+
+        return self.pymc_node(func, 
+                              kwargs['doc'], 
+                              name, 
+                              parents = parents,
+                              trace = self.keep_regressor_trace)
+
+class HDDMnnRegressor(HDDM):
```

Comment:
Can't you inherit from `HDDMRegressor` rather than copying it?

Reply:
done